src/device-state.coffee

```diff
@@ -241,17 +241,20 @@ module.exports = class DeviceState extends EventEmitter
 		Promise.using @_inferStepsLock, -> fn()
 
 	setTarget: (target) ->
-		validateState(target)
-		.then =>
-			@usingWriteLockTarget =>
-				# Apps, deviceConfig, dependent
-				@db.transaction (trx) =>
-					Promise.try =>
-						@config.set({ name: target.local.name }, trx)
-					.then =>
-						@deviceConfig.setTarget(target.local.config, trx)
-					.then =>
-						@applications.setTarget(target.local.apps, target.dependent, trx)
+		Promise.join(
+			@config.get('resinApiEndpoint'),
```

Comment:
This is a good opportunity to just replace this with `apiEndpoint` and remove the resinApiEndpoint config field, since it's legacy and not needed anymore.

Reply:
I'd rather handle that in a seperate PR, just to keep the changes smaller and more focussed.